src/model/mosaic/MosaicNames.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NamespaceId } from '../namespace/NamespaceId';
+import { MosaicId } from './MosaicId';
+
+/**
+ * Mosaic with linked names
+ */
+export class MosaicNames {
+
+    /**
+     *
+     */
+    constructor(
+        /**
+         * Mosaic Id
+         */
+        public readonly mosaicId: MosaicId,
+        /**
+         * Address linked namespace names
+         */
+        public readonly names: string[]) {
```

Comment:
Consistency check: AccountNames uses a NamespaceId array, here you are using a string array. 

Both must be consistent and probably should include `names` values not only namespace ids (see AccountNames PR comment).

Reply:
Fixed